packages/loader/container-utils/src/error.ts

```diff
@@ -180,11 +185,34 @@ export class DataProcessingError extends LoggingError implements IErrorBase, IFl
     }
 }
 
-export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMessage) => ({
-    messageClientId: message.clientId,
-    messageSequenceNumber: message.sequenceNumber,
-    messageClientSequenceNumber: message.clientSequenceNumber,
-    messageReferenceSequenceNumber: message.referenceSequenceNumber,
-    messageMinimumSequenceNumber: message.minimumSequenceNumber,
-    messageTimestamp: message.timestamp,
-});
+export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMessage) => {
+    const baseTelemetry = {
+        messageClientId: message.clientId,
+        messageSequenceNumber: message.sequenceNumber,
+        messageClientSequenceNumber: message.clientSequenceNumber,
+        messageReferenceSequenceNumber: message.referenceSequenceNumber,
+        messageMinimumSequenceNumber: message.minimumSequenceNumber,
+        messageTimestamp: message.timestamp,
+    };
+
+    switch (message.type) {
+        case MessageType.ClientJoin: {
+            const systemJoinMessage = message as ISequencedDocumentSystemMessage;
```

Comment:
If I have to guess, protocol implementation was throwing some error due to duplicate join ops. If so, I'd rather see protocol throwing error that is properly annotated with this info, and not put it here. It feels too specific of very specific code flow - if we keep doing similar changes here for other cases, this function becomes really long.

Looking through the code, it's likely 0x1ce assert that needs to be changed to throw proper error with clientId as one of the properties. Same for 0x1cf

Reply:
Are you saying do something like `throw new Error(Cannot remove quorum member. Cliend ID ${someId} not found.)` in protocol, and still let DeltaManager decide what FluidError needs to be created out of that? It may be worthwhile to be more explicit on the FluidError type at the point where error occurs.

Another reason Im asking is that this quorum error did translate to DataCorruptionError for 1.2.2 loader (one of the livesite OCE issues was around that). But now looking at that our main and changes we have done there, it seems that this becomes DataProcessingError. DataProcessingError "may be a transient error" per definition. Maybe some clarity there ? Can we really always be conclusive (or mostly conclusive) that particular error condition in our logic was transient or not ? For that particular issue, it was server who generated multiple join/leaves, and the error was not transient in nature as the dup op was closing the container.